File: suivi-scolaire-projet-complet/mobile-android/app/src/main/java/com/ecole/suiviscolaire/data/model/Annonce.kt
package com.ecole.suiviscolaire.data.model

import com.google.gson.annotations.SerializedName

data class Annonce(
    @SerializedName("id") val id: Int,
    @SerializedName("titre") val titre: String,
    @SerializedName("contenu") val contenu: String,
    @SerializedName("type") val type: String, // "annonce" ou "notification"
    @SerializedName("classe") val classe: String?,
    @SerializedName("date_publication") val datePublication: String?
) {
    val estNotification: Boolean get() = type == "notification"
}

File: suivi-scolaire-projet-complet/backend-laravel/app/Models/Annonce.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Annonce extends Model
{
    protected $fillable = [
        'titre',
        'contenu',
        'type',
        'classe_id',
        'date_publication',
    ];

    protected $casts = [
        'date_publication' => 'datetime',
    ];

    public function classe()
    {
        return $this->belongsTo(Classe::class);
    }
}
